docs/tutorials/basic/exception_handling.md

```diff
@@ -0,0 +1,111 @@
+number of samples. This should throw an exception. 
+
+The CSVIter uses the PrefetcherIter for loading and parsing data. 
+The PrefetcherIter spawns a producer thread in the background which prefetches
+the data while the main thread consumes the data. The exception is thrown in the spawned
+producer thread during the prefetching, when the label is not found corresponding to a specific sample. 
+
+The exception is transported to the main thread, where it is rethrown when Next is 
+called as part of the following line: `for batch in iter(data_train)`.
+
+In general, Exception may be rethrown as part of `Next` and `BeforeFirst` calls which correspond to `reset()` and `next()` methods in `MXDataIter` for Python language bindings.
+
+```python
+import os
+import mxnet as mx
+
+cwd = os.getcwd()
+data_path = os.path.join(cwd, "data.csv")
+label_path = os.path.join(cwd, "label.csv")
+
+with open(data_path, "w") as fout:
+    for i in range(8):
+        fout.write("1,2,3,4,5,6,7,8,9,10\n")
+
+with open(label_path, "w") as fout:
+    for i in range(7):
+        fout.write("label"+str(i))
+
+try:
+    data_train = mx.io.CSVIter(data_csv=data_path, label_csv=label_path, data_shape=(1, 10),
+                               batch_size=4)
+
+    for batch in iter(data_train):
+        print(data_train.getdata().asnumpy())
+except mx.base.MXNetError as ex:
+    print("Exception handled")
+    print(ex)
+```
+
+### Limitation
+
+There is a race condition when your last `next()` call doesnt reach the batch in your dataset where exception occurs. Exception may or may not be thrown in this case depending on which thread wins the race. To avoid this situation, you should try and iterate through your full dataset if you think it can throw exceptions which need to be handled.
+
+
+## Exception Handling for Operators
+
+The below example shows how to handle exceptions for operators in the imperative mode.
+
+For the operator case, the dependency engine spawns a number of threads if it is running in the `ThreadedEnginePool` or `ThreadedEnginePerDevice` mode. The final operator is executed in one of the spawned threads. 
+
+If an operator throws an exception during execution, this exception is propagated
+down the dependency chain. Once there is a synchronizing call i.e. WaitToRead for a variable, in the dependency chain, the propagated exception is rethrown. 
+
+In the below example, I illustrate how an exception that occured in the first line is propagated down the dependency chain, and finally is rethrown when we make a synchronizing call to WaitToRead.
+
+```python
+import mxnet as mx
+a = mx.nd.random.normal(0, 1, (2, 2))
+b = mx.nd.random.normal(0, 2, (2, 2))
+c = mx.nd.dot(a, b)
+d = mx.nd.random.normal(0, -1, (2, 2))
+e = mx.nd.dot(c, d)
+e.wait_to_read()
+```
+
+Although, the above exception occurs when executing the operation which writes to the variable d in one of the child threads, it surfaces only when the synchronizing `wait_to_read`.
```

Comment:
remove comma after although
variable `d`
rephrase 'when the synchronizing `wait_to_read'